src/timedelta_isoformat/__init__.py

```diff
@@ -1,19 +1,35 @@
 """Supplemental ISO8601 duration format support for :py:class:`datetime.timedelta`"""
 import datetime
-from typing import Iterable, Tuple, TypeAlias
+from typing import Iterable, Tuple, TypeAlias, Dict
+from dataclasses import dataclass
 
 _DIGITS, _DECIMAL_SIGNS = frozenset("0123456789"), frozenset(",.")
 _FORMAT = _DIGITS | _DECIMAL_SIGNS
 
 
+Token: TypeAlias = str
+TimedeltaArgument: TypeAlias = str
+UnparsedValue: TypeAlias = str
+ValueLimit: TypeAlias = int | None
+Components: TypeAlias = Iterable[Tuple[UnparsedValue, TimedeltaArgument, ValueLimit]]
+Measurements: TypeAlias = Iterable[Tuple[TimedeltaArgument, float]]
+
+class ParsingContext:
+    unparsed_valid_tokens: Iterable[Token]
+    token_to_timedelta_arg: Dict[Token, TimedeltaArgument]
+
+    def __init__(
+        self,
+        token_to_timedelta_arg: Dict[Token, TimedeltaArgument],
+    ):
+        self.token_to_timedelta_arg = token_to_timedelta_arg
+        self.unparsed_valid_tokens = iter(token_to_timedelta_arg)
+
 class timedelta(datetime.timedelta):
     """Subclass of :py:class:`datetime.timedelta` with additional methods to implement
     ISO8601-style parsing and formatting.
     """
 
-    Components: TypeAlias = Iterable[Tuple[str, str, int | None]]
```

Comment:
I hadn't encountered TypeAlias before and had just been doing `Token = str`. Now I can't get enough!